augur/util_support/auspice_config.py

```diff
@@ -0,0 +1,234 @@
+        for (idx, base_el) in enumerate(values):
+            if identity_func(overlay_el, base_el):
+                values[idx] = overlay_el # replace existing element with one from the overlay config
+                break
+        else:
+            values.append(overlay_el)
+
+    return {**base, config_key: values}
+
+def _merge_scalar(base: dict[str, Any], overlay: dict[str, Any], config_key: str) -> dict[str, Any]:
+    return {**base, config_key: overlay[config_key]} if config_key in overlay else base
+
+def _merge_dicts(base: dict[str, Any], overlay: dict[str, Any], config_key: str) -> dict[str, Any]:
+    """
+    Dicts are _not_ recursively merged, they are simply merged by adding new keys from the overlay to
+    those keys already present in the base. Identical keys replace the original key.
+    """
+    if config_key not in overlay:
+        return base
+    if config_key not in base:
+        return {**base, config_key: overlay[config_key]}
+
+    if not isinstance(overlay[config_key], dict) or not isinstance(base[config_key], dict):
+        raise AugurError(f"Config merging for {config_key!r} failed as one (or more) entries were not dictionaries")
+
+    return {**base, config_key: {**base[config_key], **overlay[config_key]}}
+
+def _geo_resolution_id(el: Union[str, dict[str,str]]) -> str:
+    return el['key'] if isinstance(el, dict) else el
+
+def _replace_deprecated(config: dict[str, Any], old_name: str, new_name: Union[None,str], modify: Union[None, Callable]=None):
+    if old_name not in config:
+        return # NO-OP
+    elif new_name is None:
+        deprecated(f"[config file] key {old_name!r} is no longer used and has been dropped from your config")
+        del config[old_name]
+    elif old_name == new_name: # indicates deprecations within an unchanged top-level key
+        assert modify is not None, "[internal error] modify must be a function"
+        config[new_name] = modify(config[old_name])
+    elif new_name in config:
+        deprecated(f"[config file] top level keys {new_name!r} and {old_name!r} were both present, " \
+                   "however the latter is a deprecated version of the former and will be ignored.")
+        del config[old_name]
+    else:
+        deprecated(f"[config file] top level key {old_name!r} is deprecated. Renaming to {new_name!r}" +
+                   (" (with modifications)" if modify else ""))
+        config[new_name] = modify(config[old_name]) if modify else config[old_name]
+        del config[old_name]
+
+def _parse_color_options(options: Any) -> list[dict]:
+    # parse v1-style 'color_options' & convert to v2-style 'colorings'
+    if not isinstance(options, dict):
+        raise AugurError(f"[config file] The deprecated field 'color_options' must be a dictionary")
+    colorings = []
+    for key, info in options.items():
+        # note: if both legentTitle & menuItem are present then we use the latter. See https://github.com/nextstrain/auspice/issues/730
```

Comment:
```suggestion
        # note: if both legendTitle & menuItem are present then we use the latter. See https://github.com/nextstrain/auspice/issues/730
```

Reply:
Good eyes - this typo wasn't introduced by me (it exists in current master) but I've fixed it up